### DISCUSSION TITLE
JsPsych Preload Plugin Support

### DISCUSSION THREAD

**Discussion by Tanush99** [jspsych/jsPsych#2511] [Q&A]
Hello JsPsych Team,

Thanks for the JsPsych framework. I just had a small question regarding the JsPsych Preload plugin. Is the plugin currently supported? This is because whenever I use a function such as:

var preload = {
    type: 'preload',
    auto_preload: true,
}

I am receiving errors stating how jsPsych.getSafeModeStatus is not a function even though it seems to be called in the jspsych-preload.js file. 

When I remove the line calling jsPsych.safeModeStatus, I seem to be receiving another error stating how jsPsych.pluginAPI.getAutoPreloadList is not a function. Therefore, is this plugin currently supported since this plugin seems to be missing functions? Or is there an updated version of this plugin. I received this jspsych-preload.js file from the v.6.3.1 JsPsych distribution.

Thanks

**Comment by becky-gilbert** [comment]
Hi @Tanush99, yes the `preload` plugin was added in v6.3.0 and has been supported since then. It sounds like there might be a mismatch in the jsPsych version you're using and the preload plugin version. Do you know what jsPsych version you're using?
